apiserver/cmd/main.go

```diff
@@ -144,14 +144,24 @@ func startHttpProxy() {
 	topMux.HandleFunc("/healthz", serveHealth)
 	serveSwaggerUI(topMux)
 
-	if err := http.ListenAndServe(*httpPortFlag, topMux); err != nil {
+	// Create a custom HTTP server with timeouts.
+	srv := &http.Server{
```

Comment:
linter requires timeout for http server operations; we could have a followup issue to add proper timeout value, use 0 to disable timeout to keep existing behavior.